data/test/scenarios/wml_tests/UnitsWML/zero_experience.cfg

```diff
@@ -0,0 +1,76 @@
+# wmllint: no translatables
+
+#####
+# API(s) being tested: [unit] max_experience=0
+##
+# Actions:
+# Change Alice's and Bob's max_experience to 0. Delete Bob's 'Orcish Warrior' advancement but not Alice's. Harm bob with 1 hitpoint to be sure that he have some damage. They then fight each other.
+##
+# Expected end state:
+# Alice levels up. Bob doesn't, and keeps his previous health.
+#####
+{COMMON_KEEP_A_B_UNIT_TEST "zero_experience" (
+    [event]
+        name=prestart
+        [modify_unit]
+            [filter]
+                id=alice,bob
+            [/filter]
+            [effect]
+                apply_to=max_experience
+                set=0
+            [/effect]
+        [/modify_unit]
+        [modify_unit]
```

Comment:
At this point, `[store_unit]` and `{ASSERT ({VARIABLE_CONDITIONAL bob.max_experience numerical_equals 0})}`. That assert fails, in the stored WML Bob has `max_experience==1`.

That seems to be a sanity-check in `unit::init`, `std::max(1, v->to_int(max_experience_))`. I think it's good to have a test that documents that that sanity-check works, but the rest of this unit test isn't testing what you expect.